src/apps/chatter/hooks/useWebSockets.js

```diff
@@ -0,0 +1,92 @@
+import url from '/util/url'
+import {
+  useRef
+} from 'react'
+
+export function useWebSockets (roomId) {
+  const socketUrl = url('api_ws.chatSocket', {
+    args: {
+      id: roomId.toString()
+    }
+  })
```

Comment:
I think the naming could be improved for this. This custom hook doesn't let us connect multiple websockets, rather it let's us connect a single websocket related to a Room. I think the naming should reflect it's more singular focus, perhaps `useRoom` would do?

To be clear, I think it's better to make a more focused custom hook for our purposes (as you're doing), rather than a more generalized one. Too much abstraction (especially early on) can lead to clumsy code. Clumsy, meaning, less explicit and clear.

also:
```
      id: roomId.toString()
```

Do websockets not support numbers?